src/generic/oomph_utilities.cc

```diff
@@ -1306,6 +1306,41 @@ namespace oomph
         return double(t) / double(CLOCKS_PER_SEC);
       }
     }
+
+    /// Returns a nicely formatted string from an input time in seconds;
+    /// the format depends on the size of time, e.g.:
+    ///  3710 will be printed as 1h:01:50
+    ///   700 will be printed as 11m:40
+    ///    59 will be printed as 59s
+    std::string format_time(const double& time)
```

Comment:
Might be more meaningful to name this something like `get_time_as_string(...)`

Reply:
Also, I think the input should be something like a `long long unsigned` or `uint64_t` (after adding `#include <cstdint>`) given that you don't handle quantities smaller than a single second (i.e. `1`)

Reply:
agreed